packages/typespec-test/test/openai_modular/generated/typespec-ts/src/api/operations.ts

```diff
@@ -460,105 +400,63 @@ export async function _getChatCompletionsWithAzureExtensionsDeserialize(
     throw createRestError(result);
   }
 
-  return {
-    id: result.body["id"],
-    created: new Date(result.body["created"]),
-    choices: result.body["choices"].map((p) => ({
-      message: !p.message ? undefined : (p.message as any),
```

Comment:
I am not sure if removing this would work, if I remember corrrectly, there's a recursive reference here. which has not yet been resolved. https://github.com/Azure/autorest.typescript/issues/1982 

/cc @dgetu

Reply:
If the recursive properties don't need to be reshaped it should be fine, as we'll be returning the same as we got from the service. True, this doesn't handle recursive properties that need to be transformed.